### DISCUSSION TITLE
Support passing functions in AG Grid GridOptions

### DISCUSSION THREAD

**Discussion by PyAntony** [zauberzeug/nicegui#1420] [Ideas / Feature Requests]
### Question

Hi,

GridOptions has many parameters that are functions (not just text and booleans). For example:
![image](https://github.com/zauberzeug/nicegui/assets/24689636/df2c8d7e-d210-4816-b6d0-9da57f332c47)

I am trying to add an external filter. But this doesn't work:
```py
grid = ui.aggrid.from_pandas(dataframe)
grid.options['isExternalFilterPresent'] = lambda params: True
```

How can I assign functions?? 

If not possible, will it be possible in the future?

**Comment by falkoschindler** [comment]
Hi @PyAntony, currently you can't pass functions to the options of `ui.aggrid`. I hope we can add this functionality in the future, like [we did for `ui.table`](https://nicegui.io/documentation/table#custom_sorting_and_formatting).

I'll convert this discussion into a feature request. Maybe someone from the community has an idea for the implementation?

Here is an example how the API could look like:
```py
ui.aggrid({
    'columnDefs': [
        {'headerName': 'Name', 'field': 'name'},
        {'headerName': 'Age', 'field': 'age'},
    ],
    'rowData': [
        {'name': 'Alice', 'age': 18},
        {'name': 'Bob', 'age': 21},
        {'name': 'Carol', 'age': 42},
    ],
    'getRowStyle': '(params) => params.data.age < 25 ? { backgroundColor: "red" } : null',
})
```

**Reply by PyAntony** [reply]
This would be great. Thanks. I was thinking passing a class my be better, something like: 
```
'getRowStyle': JavaScriptFunction(params, somethingElse...)
```
That might add type safety and maybe easier to parse/execute in the library? (no idea how nicegui internals works).

**Reply by jackiechangmm** [reply]
It's really needed! otherwise aggrid wouldn't be able to perform its powerful.

**Comment by falkoschindler** [comment]
This feature has been released in version 1.3.12.

**Reply by PyAntony** [reply]
Is documentation updated with an example? If not could you please share an example here? Thanks.

**Reply by rodja** [reply]
The https://nicegui.io/documentation/aggrid#ag_grid_with_dynamic_row_height demo is now live.
